query/src/sql/exec/data_schema_builder.rs

```diff
@@ -80,6 +81,27 @@ impl<'a> DataSchemaBuilder<'a> {
         Ok(Arc::new(DataSchema::new(fields)))
     }
 
+    pub fn build_expression_plan(
+        &self,
+        exprs: &[Expression],
+        input_schema: DataSchemaRef,
+    ) -> Result<DataSchemaRef> {
+        let mut fields = input_schema.fields().clone();
+        for expr in exprs.iter() {
+            let expr_name = expr.column_name().clone();
+            if input_schema.has_field(expr_name.as_str()) {
+                continue;
+            }
+            let field = if expr.nullable(&input_schema)? {
```

Comment:
Maybe better to use match when we care about all branches and the get results from branches.